File: expert/src/main/java/com/qing/expert/entity/Order.java
package com.qing.expert.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("orders")
public class Order extends BaseEntity {

    /**
     * 订单编号
     */
    @TableField("order_no")
    private String orderNo;

    /**
     * 用户ID
     */
    @TableField("user_id")
    private Long userId;

    /**
     * 达人ID
     */
    @TableField("expert_id")
    private Long expertId;

    /**
     * 服务ID
     */
    @TableField("service_id")
    private Long serviceId;

    /**
     * 服务名称
     */
    @TableField("service_name")
    private String serviceName;

    /**
     * 服务价格
     */
    @TableField("service_price")
    private BigDecimal servicePrice;

    /**
     * 订单标题
     */
    @TableField("title")
    private String title;

    /**
     * 订单描述
     */
    @TableField("description")
    private String description;

    /**
     * 订单金额
     */
    @TableField("amount")
    private BigDecimal amount;

    /**
     * 实付金额
     */
    @TableField("paid_amount")
    private BigDecimal paidAmount;

    /**
     * 优惠金额
     */
    @TableField("discount_amount")
    private BigDecimal discountAmount;

    /**
     * 订单状态：1-待接单，2-已接单，3-服务中，4-已完成，5-已取消，6-售后中
     */
    @TableField("status")
    private Integer status;

    /**
     * 支付状态：0-未支付，1-已支付，2-已退款
     */
    @TableField("pay_status")
    private Integer payStatus;

    /**
     * 支付方式：WECHAT,BALANCE
     */
    @TableField("pay_type")
    private String payType;

    /**
     * 支付时间
     */
    @TableField("pay_time")
    private LocalDateTime payTime;

    /**
     * 期望完成时间
     */
    @TableField("expected_time")
    private LocalDateTime expectedTime;

    /**
     * 接单时间
     */
    @TableField("accept_time")
    private LocalDateTime acceptTime;

    /**
     * 开始服务时间
     */
    @TableField("start_time")
    private LocalDateTime startTime;

    /**
     * 完成时间
     */
    @TableField("finish_time")
    private LocalDateTime finishTime;

    /**
     * 取消时间
     */
    @TableField("cancel_time")
    private LocalDateTime cancelTime;

    /**
     * 取消原因
     */
    @TableField("cancel_reason")
    private String cancelReason;

    /**
     * 用户备注
     */
    @TableField("user_remark")
    private String userRemark;

    /**
     * 达人备注
     */
    @TableField("expert_remark")
    private String expertRemark;

    /**
     * 管理员备注
     */
    @TableField("admin_remark")
    private String adminRemark;

    /**
     * 联系方式
     */
    @TableField("contact_info")
    private String contactInfo;

    /**
     * 服务地址
     */
    @TableField("service_address")
    private String serviceAddress;

    /**
     * 预约时间
     */
    @TableField("appointment_time")
    private LocalDateTime appointmentTime;
}

File: uniapp/Uni_expert/api/order.ts
/**
 * 订单相关API
 */
import request from './request'
import type { PageResponse } from './request'

// 订单状态枚举（与数据库表一致）
export enum OrderStatus {
  PENDING = 1,     // 待接单
  ACCEPTED = 2,    // 已接单
  IN_SERVICE = 3,  // 服务中
  COMPLETED = 4,   // 已完成
  CANCELLED = 5,   // 已取消
  AFTER_SALE = 6   // 售后中
}

// 支付状态枚举
export enum PaymentStatus {
  UNPAID = 0,    // 未支付
  PAID = 1,      // 已支付
  REFUNDED = 2   // 已退款
}

// 订单接口
export interface Order {
  id: number
  orderNo: string
  userId: number
  expertId?: number  // 修正：talentId -> expertId
  serviceId: number  // 添加：服务ID
  serviceName: string
  servicePrice: number
  title?: string     // 添加：订单标题
  description: string // 修正：taskDescription -> description
  amount: number     // 添加：订单金额
  expectedTime?: string
  status: OrderStatus    // 修正：orderStatus -> status
  payStatus: PaymentStatus // 修正：paymentStatus -> payStatus
  payType?: string   // 添加：支付方式
  payTime?: string   // 添加：支付时间
  acceptTime?: string
  startTime?: string // 添加：开始服务时间
  finishTime?: string // 修正：completeTime -> finishTime
  cancelTime?: string
  cancelReason?: string // 添加：取消原因
  userRemark?: string   // 添加：用户备注
  expertRemark?: string // 添加：达人备注
  contactInfo?: string  // 添加：联系方式
  serviceAddress?: string // 添加：服务地址
  appointmentTime?: string // 添加：预约时间
  createTime: string
  updateTime?: string

  // 关联信息
  userInfo?: {
    id: number
    nickname: string
    avatar?: string
    phone?: string
  }
  expertInfo?: {  // 修正：talentInfo -> expertInfo
    id: number
    expertName: string // 修正：name -> expertName
    avatar?: string
    phone?: string
    rating: number
  }
  categoryInfo?: {
    id: number
    name: string
    icon?: string
  }
}

// 创建订单参数
export interface CreateOrderParams {
  serviceId: number     // 修正：使用serviceId而不是talentId
  title: string         // 添加：订单标题
  description: string   // 修正：taskDescription -> description
  amount: number        // 添加：订单金额
  userRemark?: string   // 添加：用户备注
  contactInfo?: string  // 添加：联系方式
  serviceAddress?: string // 添加：服务地址
  appointmentTime?: string // 添加：预约时间
}

// 订单查询参数
export interface OrderQueryParams {
  current?: number
  size?: number
  status?: OrderStatus
  keyword?: string
  startTime?: string
  endTime?: string
}

// 订单评价参数
export interface OrderReviewParams {
  orderId: number
  rating: number
  serviceRating?: number
  speedRating?: number
  qualityRating?: number
  content?: string
  images?: string[]
  isAnonymous?: boolean
}

// 创建订单
export const createOrder = (params: CreateOrderParams): Promise<{ orderId: number }> => {
  return request.post('/orders', params)
}

// 获取订单列表
export const getOrderList = (params: OrderQueryParams): Promise<PageResponse<Order>> => {
  return request.get('/orders', params)
}

// 获取用户订单列表
export const getUserOrders = (params: OrderQueryParams): Promise<PageResponse<Order>> => {
  return request.get('/user/orders', params)
}

// 获取达人订单列表
export const getExpertOrders = (params: OrderQueryParams): Promise<PageResponse<Order>> => {
  return request.get('/expert/orders', params)
}

// 获取订单详情
export const getOrderDetail = (orderId: number): Promise<Order> => {
  return request.get(`/orders/${orderId}`)
}

// 取消订单
export const cancelOrder = (orderId: number, reason?: string): Promise<void> => {
  return request.put(`/orders/${orderId}/cancel`, { reason })
}

// 确认订单
export const confirmOrder = (orderId: number): Promise<void> => {
  return request.put(`/orders/${orderId}/confirm`)
}

// 达人接单
export const acceptOrder = (orderId: number): Promise<void> => {
  return request.put(`/orders/${orderId}/accept`)
}

// 开始服务
export const startService = (orderId: number): Promise<void> => {
  return request.put(`/orders/${orderId}/start`)
}

// 完成订单
export const completeOrder = (orderId: number, data: {
  completionScreenshots?: string[]
  talentSignature?: string
}): Promise<void> => {
  return request.put(`/orders/${orderId}/complete`, data)
}

// 评价订单
export const reviewOrder = (params: OrderReviewParams): Promise<void> => {
  return request.post('/orders/review', params)
}

// 申请售后
export const applyAfterSale = (orderId: number, reason: string): Promise<void> => {
  return request.post(`/orders/${orderId}/after-sale`, { reason })
}

// 获取订单状态文本
export const getOrderStatusText = (status: OrderStatus): string => {
  const statusMap = {
    [OrderStatus.PENDING]: '待接单',
    [OrderStatus.ACCEPTED]: '已接单',
    [OrderStatus.IN_SERVICE]: '服务中',
    [OrderStatus.COMPLETED]: '已完成',
    [OrderStatus.CANCELLED]: '已取消',
    [OrderStatus.AFTER_SALE]: '售后中'
  }
  return statusMap[status] || '未知状态'
}

// 获取支付状态文本
export const getPaymentStatusText = (status: PaymentStatus): string => {
  const statusMap = {
    [PaymentStatus.UNPAID]: '未支付',
    [PaymentStatus.PAID]: '已支付',
    [PaymentStatus.REFUNDED]: '已退款'
  }
  return statusMap[status] || '未知状态'
}

// 获取订单状态颜色
export const getOrderStatusColor = (status: OrderStatus): string => {
  const colorMap = {
    [OrderStatus.PENDING]: '#f0ad4e',
    [OrderStatus.ACCEPTED]: '#5bc0de',
    [OrderStatus.IN_SERVICE]: '#007aff',
    [OrderStatus.COMPLETED]: '#4cd964',
    [OrderStatus.CANCELLED]: '#999999',
    [OrderStatus.AFTER_SALE]: '#ff6b35'
  }
  return colorMap[status] || '#999999'
}

// 支付订单
export const payOrder = (orderId: number, paymentMethod: string = 'wechat'): Promise<{
  paymentParams: any
}> => {
  return request.post(`/orders/${orderId}/pay`, { paymentMethod })
}
